op-service/txmgr/txmgr.go

```diff
@@ -124,7 +141,8 @@ type SimpleTxManager struct {
 	nonce     *uint64
 	nonceLock sync.RWMutex
 
-	pending atomic.Int64
+	numPending atomic.Int64
+	pendingTxs map[uint64]*PendingTxWithCancel
```

Comment:
Does this data structure provide everything we need? Namely, does the assumption that there will only ever be one pendingTx per nonce hold true? Or would we want to nest a map or slice to allow for multiple in-flight pending txs for a single nonce, e.g.: `pendingTxs map[uint64]map[common.Hash]*PendingTxWithCancel`

Reply:
Also, do we need a mutex to protect R/W of the `pendingTxs`?